starters/next-wordpress-starter/pages/pages/[uri].jsx

```diff
@@ -17,9 +19,10 @@ export default function PageListTemplate({ menuItems, page }) {
   );
 }
 
-export async function getServerSideProps({ params: { uri } }) {
+export async function getServerSideProps({ params: { uri }, res }) {
   const menuItems = await getFooterMenu();
   const page = await getPageByUri(uri);
+  setEdgeHeader({ res });
```

Comment:
Looks like we can add a headers option to the GraphQLClient and it will send that header with every request.

https://github.com/prisma-labs/graphql-request#passing-dynamic-headers-to-the-client

Reply:
@stevector and @CobyPear I agree that it would be better to set this globally. But it looked to be harder than I originally assumed when this was scoped out. From my PR description:

> My initial intent here was to override elements of graphql-request to do this, similar to what we're doing with Drupal State. This ended up being a larger effort than I expected, partially because we don't maintain graphql-request. That's why I ended up with this approach that requires instrumenting setEdgeHeaders. We could revisit this more-hands off approach in the future, but seeing this in action I wonder if the clarity that this simplified approach provides is actually a benefit...

I'm assuming here that we don't want a period of time where we have SSR for most routes but no edge caching, hence the short term fix. Going with the short term fix would be my preference, but we could hold this for a long term solution if preferred.

I also tried both of the solutions suggested above, and don't believe they can apply here.

@stevector:

> rather than needing to call setEdgeHeader on every single route, can we set this globally?

https://nextjs.org/docs/api-reference/next.config.js/headers

From https://nextjs.org/docs/api-reference/next.config.js/headers#cache-control:

> You cannot set Cache-Control headers in next.config.js file as these will be overwritten in production to ensure that API Routes and static assets are cached effectively.

That's why currently we're setting this in getServerSide props via a helper.

@CobyPear:

> Looks like we can add a headers option to the GraphQLClient and it will send that header with every request.

https://github.com/prisma-labs/graphql-request#passing-dynamic-headers-to-the-client

I also tried this, but that is setting the wrong headers. That will set the headers on the client requests to wp-graphql - we need to set the header on Next.js's outgoing response.

My best current guess at a longer term solution is to extend graphql-request so that we can pass the node response object and set the appropriate headers if we make WordPress API calls. That is similar to what we're currently doing for Drupal, but tbh it is still really just moving the responsibility around - developers need to use our data sourcing utility and pass the response object correctly. For that reason I think there could still be a use for the utility introduced here - especially for customers who want to use a different approach to querying wp-graphql.

We'll face similar but even more complex problems when we need to start bubbling up surrogate keys to the outgoing response.. It will likely require a similar solution.